File: Ejers 700 - 799/p739.py
import math

def calcular_minimo_azulejos(casos):
    resultados = []
    for caso in casos:
        W, H, w, h = caso
        # Colocación en horizontal
        azulejos_horizontal = math.ceil(W / w) * math.ceil(H / h)
        print("htz",azulejos_horizontal)
        # Colocación en vertical
        azulejos_vertical = math.ceil(W / h) * math.ceil(H / w)
        print("vtr",azulejos_vertical)
        # El mínimo de ambas colocaciones
        resultados.append(min(azulejos_horizontal, azulejos_vertical))
    return resultados

# Ejemplo de entrada
casos = [
    (4, 3, 2, 1),
    (3, 4, 2, 1),
    (11, 7, 5, 2)
]

# Procesar los casos de prueba
resultados = calcular_minimo_azulejos(casos)
for resultado in resultados:
    print(resultado)

File: Ejers 700 - 799/p739.java
import java.util.*;

public class p739 {
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);

        int t = s.nextInt();

        while (t-- != 0) {
            int an = s.nextInt();
            int An = s.nextInt();
            int al = s.nextInt();
            int Al = s.nextInt();

            int hrz = (int) (Math.ceil((double) an / al) * Math.ceil((double) An / Al));
            int vrt = (int) (Math.ceil((double) an / Al) * Math.ceil((double) An / al));

            System.out.println(Math.min(hrz,vrt));
        }
    }
}
